libvast/vast/table_slice.hpp

```diff
@@ -301,6 +301,27 @@ table_slice_ptr truncate(const table_slice_ptr& slice, size_t num_rows);
 std::pair<table_slice_ptr, table_slice_ptr> split(const table_slice_ptr& slice,
                                                   size_t partition_point);
 
+/// Counts the number of total rows of multiple table slices.
+/// @param slices The table slices to count.
+/// @returns The sum of rows across *slices*.
+uint64_t rows(const std::vector<table_slice_ptr>& slices);
+
+/// Converts the table slice into a 2-D matrix in row-major order such that
+/// each row represents an event.
+/// @param slice The table slice to convert.
+/// @param first_row An offset to the first row to consider.
+/// @param num_rows Then number of rows to consider. (0 = all rows)
+/// @returns a 2-D matrix of data instances corresponding to *slice*.
+/// @requires first_row < slice->rows()
+/// @requires num_rows <= slice->rows() - first_row
+/// @note This function exists primarily for unit testing because it performs
+/// excessive memory allocations.
+std::vector<std::vector<data>>
+to_data(const table_slice& slice, size_t first_row = 0, size_t num_rows = 0);
```

Comment:
The two arguments seem to be unused in this PR, I assume this will be necessary in a later commit?

Reply:
Right, these are used in unit tests (which i fixed up later).